client/node_updater.go

```diff
@@ -123,6 +132,25 @@ func (c *Client) updateNodeFromCSI(name string, info *structs.CSIInfo) {
 	}
 }
 
+func (c *Client) updateNodeFromHostVol(name string, vol *structs.ClientHostVolumeConfig) {
+	c.configLock.Lock()
+	defer c.configLock.Unlock()
+
+	newConfig := c.config.Copy()
+
+	if newConfig.Node.HostVolumes == nil {
+		newConfig.Node.HostVolumes = make(map[string]*structs.ClientHostVolumeConfig) // TODO: why?
```

Comment:
Generally speaking the various maps in configs are `nil` and not empty if unset (I suppose so that we're not allocating the [internal structs for the map](https://dave.cheney.net/2018/05/29/how-the-go-runtime-implements-maps-efficiently-without-generics), although admittedly that doesn't seem like a lot of memory overall). If there's no static host volume, the first time we access this it would be an assignment-to-nil-map panic. So we need to initialize the map on first use.